awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/UdpSender.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.opentelemetry.javaagent.providers;
+
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetSocketAddress;
+import java.net.SocketException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class represents a UDP sender that sends data to a specified endpoint. It is used to send
+ * data to a remote host and port using UDP protocol.
+ */
+class UdpSender {
+  private static final Logger logger = Logger.getLogger(UdpSender.class.getName());
+
+  private DatagramSocket socket;
+  private final InetSocketAddress endpoint;
+
+  public UdpSender(String host, int port) throws SocketException {
+    this.endpoint = new InetSocketAddress(host, port);
+    try {
+      this.socket = new DatagramSocket();
+    } catch (SocketException e) {
+      logger.log(Level.SEVERE, "Exception while instantiating UdpSender socket.", e);
+    }
+  }
+
+  public CompletableResultCode shutdown() {
+    try {
+      socket.close();
```

Comment:
1. `socket` can be null if it fails in constructor with SocketException. You might want to check it to avoid NPE.
2. Since we've caught `SocketException`, is it still needed to claim `throws SocketException` at constructor?

Reply:
Done. No we don't need the `throws SocketException` in the constructor signature.